src/ergw_aaa_diameter_srv.erl

```diff
@@ -429,11 +429,37 @@ filter_avp(AVPs, _) ->
     AVPs.
 
 check_avp_conditions(AVPs, [{AVP, Value} | Rest]) when is_map_key(AVP, AVPs) ->
-    case maps:get(AVP, AVPs) of
-	Value -> check_avp_conditions(AVPs, Rest);
+    case match_avp_value(Value, maps:get(AVP, AVPs)) of
+	true -> check_avp_conditions(AVPs, Rest);
 	_ -> false
     end;
 check_avp_conditions(_, []) ->
     true;
 check_avp_conditions(_, _) ->
     false.
+
+%% Since the implementation of not erlang native (json, yaml) configuration 
+%% format, we can not expect that the condition values will be matching the
+%% possible OTP diameter value formats. Some of the "variable" formats
+%% not likely to be used in matching (e.g. Time) are not implemented
+match_avp_value(ConditionVal, [AVPVal]) ->
+% OTP diameter puts message optional AVPs in a single element list
+    match_avp_value(ConditionVal, AVPVal);
+match_avp_value(ConditionVal, ConditionVal) ->
+    true;
+match_avp_value(ConditionVal, AVPVal)
+  when is_binary(ConditionVal) andalso is_integer(AVPVal) ->
+    AVPVal == binary_to_integer(ConditionVal);
+match_avp_value(ConditionVal, AVPVal)
+  when is_binary(ConditionVal) andalso is_atom(AVPVal) ->
+    AVPVal == binary_to_existing_atom(ConditionVal);
```

Comment:
Looks like better use `erlang:binary_to_existing_*/1` in all places for support all OTP versions.

Reply:
Yep, CT error:
```erlang
src/ergw_aaa_diameter_srv.erl:455: function binary_to_existing_atom/1 undefined
```

Reply:
indeed 22 only supports binary_to_existing_atom/2, fixed